tests/unit/Actions/RenderAdminNoticeTest.php

```diff
@@ -94,4 +99,99 @@ public function testShouldRenderCallbackOutput(): void
             $renderAdminNotice($notice)
         );
     }
+
+    public function testShouldIncludeAttributesForCustomNotices()
+    {
+        $notice = (new AdminNotice('test_id', function () {
+            return '<div>It is Stellar at StellarWP</div>';
+        }))
+            ->custom();
+        $expectedNoticeWithAttributes = "<div data-stellarwp-namespace-notice-id='test_id' data-stellarwp-namespace-location='below_header'>It is Stellar at StellarWP</div>";
+
+        $renderAdminNotice = new RenderAdminNotice('namespace');
+
+        $tagProcessorMock = $this->createMock(TagProcessorMock::class);
+
+        $tagMatcher = $this->exactly(2);
+        $tagProcessorMock
+            ->expects($tagMatcher)
+            ->method('set_attribute')
+            ->willReturnCallback(function (string $key, string $value) use ($tagMatcher) {
+                switch ($tagMatcher->getInvocationCount()) {
+                    case 1:
+                        $this->assertEquals('data-stellarwp-namespace-notice-id', $key);
+                        break;
+                    case 2:
+                        $this->assertEquals('data-stellarwp-namespace-location', $key);
+                        $this->assertEquals('below_header', $value);
+                        break;
+                }
+            });
+
+        $tagProcessorMock
+            ->expects($this->once())
+            ->method('next_tag');
+
+        $tagProcessorMock
+            ->expects($this->once())
+            ->method('__toString')
+            ->willReturn($expectedNoticeWithAttributes);
+
+        $this->setClassMock('WP_HTML_Tag_Processor', $tagProcessorMock);
+
+        $this->assertEquals($expectedNoticeWithAttributes, $renderAdminNotice($notice));
```

Comment:
This could be shortened and require less logic if you were to use snapshot testing (and require less rework when we change attributes):

```php
$notice = (new AdminNotice('test_id', function () {
    return '<div>It is Stellar at StellarWP</div>';
}))
    ->custom();

$renderAdminNotice = new RenderAdminNotice('namespace');
$this->assertMatchesStringSnapshot( $renderAdminNotice( $notice ) );
```

This would necessitate the following composer package: `lucatume/codeception-snapshot-assertions` and `Asserts` added to the unit.suite.yml file ([here's an example](https://github.com/stellarwp/pup/blob/main/tests/cli.suite.dist.yml#L4))

`pup` uses snapshot testing all over the place to capture CLI output, but it is great for arbitrary string comparisons that we want to persist over time. [Here's an example](https://github.com/stellarwp/pup/blob/main/tests/cli/Commands/CleanCest.php) in `pup`.

Reply:
I'd like your help with this since I haven't used this before. For now, I'll leave this test as it is working and does the job, and will look forward to refactoring it later with you! 😄